syntax = "proto3";

package faceRecognition;

message Rect {
  int32 x = 1;
  int32 y = 2;
  int32 width = 3;
  int32 height = 4;
}

message ImageRequest {
  bytes image = 1;  // We will encode our image as bytes
}

message MultipleImageRequest {
  repeated ImageRequest images = 1;  // A list of ImageRequests
}

message ImageWithRectsRequest {
  ImageRequest image = 1;  // A list of ImageRequests
  repeated Rect rects = 2;
}

message EmbeddingResponse {
  repeated float embedding = 1;  // Embedding will be a list of floats
}

message MultipleEmbeddingResponse {
  repeated EmbeddingResponse embeddings = 1;  // A list of EmbeddingResponses
}

service FaceRecognition {
  rpc getFaceEmbedding (ImageRequest) returns (EmbeddingResponse);
  rpc getFaceEmbeddings (MultipleImageRequest) returns (MultipleEmbeddingResponse);
  rpc getFaceEmbeddings2 (ImageWithRectsRequest) returns (MultipleEmbeddingResponse);
}
